### PR TITLE
Add `MessageChannel::as_either()`

### PR DESCRIPTION
**Pull request by bobdebuildr**
implements #188

### CODE REVIEW

File: src/message_channel.rs [hunk 1]
```diff
@@ -247,6 +247,20 @@ where
     }
 }
 
+/// Functions which apply to any kind of [`MessageChannel`], be they strong or weak.
```
Comment (thomaseizinger) [inline src/message_channel.rs:250]: For this comment to be true, you'd need to move away from the defaulted `Rc` variable and actually make this `impl` block generic over all `Rc`s :)

At the moment, it is only available for `Strong`.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Thanks! One comment :)

File: src/message_channel.rs (hunk 1 above)
Comment (bobdebuildr) [reply src/message_channel.rs:250]: Ah yes, thanks for catching that! I updated the `impl` block accordingly.

**Review by Restioson — APPROVED** [review]
Thanks!

**Review by thomaseizinger — APPROVED** [review]
Nice work, thanks!

**Pull request merged**